modules/apps/object/object-test/src/testIntegration/java/com/liferay/object/web/internal/portlet/action/test/ExportImportObjectFolderTest.java

```diff
@@ -0,0 +1,304 @@
+import org.junit.Before;
+import org.junit.ClassRule;
+import org.junit.Rule;
+import org.junit.Test;
+import org.junit.runner.RunWith;
+
+import org.skyscreamer.jsonassert.JSONAssert;
+import org.skyscreamer.jsonassert.JSONCompareMode;
+
+import org.springframework.http.MediaType;
+import org.springframework.mock.web.MockMultipartFile;
+import org.springframework.mock.web.MockMultipartHttpServletRequest;
+
+/**
+ * @author Guilherme Sa
+ */
+@RunWith(Arquillian.class)
+public class ExportImportObjectFolderTest {
+
+	@ClassRule
+	@Rule
+	public static final LiferayIntegrationTestRule liferayIntegrationTestRule =
+		new LiferayIntegrationTestRule();
+
+	@Before
+	public void setUp() throws Exception {
+		_user = TestPropsValues.getUser();
+
+		ObjectFolderResource.Builder builder =
+			_objectFolderResourceFactory.create();
+
+		_objectFolderResource = builder.user(
+			_user
+		).build();
+	}
+
+	@Test
+	public void testImportObjectFoldersWithObjectDefinitions()
+		throws Exception {
+
+		// Test importing and exporting an empty folder
+
+		ObjectFolder emptyFolder = _importObjectFolder(
+			"empty-folder", "empty_folder.json", "EmptyFolder");
+
+		Class<?> clazz = getClass();
+
+		JSONAssert.assertEquals(
+			StringUtil.read(
+				clazz.getResourceAsStream("dependencies/empty_folder.json")),
+			_getExportedObjectFolderJSONObject(emptyFolder.getId()),
+			JSONCompareMode.STRICT);
+
+		// Test importing a folder that has a linked object that doesn't exist
+
+		ObjectFolder objectFolder = _importObjectFolder(
+			"ObjectFolder", "object_folder.json", "ObjectFolder");
+
+		List<String> uncategorizedFolderItemERCs =
+			_getUncategorizedObjectFolderItemERCs();
+
+		Assert.assertEquals(2, objectFolder.getObjectFolderItems().length);
+		Assert.assertTrue(uncategorizedFolderItemERCs.contains("linkedERC"));
+		Assert.assertTrue(uncategorizedFolderItemERCs.contains("customERC"));
+		Assert.assertFalse(
+			uncategorizedFolderItemERCs.contains("nonExistentObjectERC"));
+
+		// Test importing a folder that moves an ObjectDefinition during import
+
+		ObjectFolder linkedFolder = _importObjectFolder(
+			"linked-object-folder", "linked_object_folder.json",
+			"LinkedObjectFolder");
+
+		uncategorizedFolderItemERCs = _getUncategorizedObjectFolderItemERCs();
+
+		Assert.assertEquals(2, linkedFolder.getObjectFolderItems().length);
+		Assert.assertFalse(uncategorizedFolderItemERCs.contains("linkedERC"));
+		Assert.assertFalse(uncategorizedFolderItemERCs.contains("customERC"));
+
+		// Test importing a folder with conflicting ERC, but with less
+		// ObjectDefinitions
+
+		_importObjectFolder(
+			"linked-object-folder", "empty_folder.json", "EmptyFolder");
+
+		uncategorizedFolderItemERCs = _getUncategorizedObjectFolderItemERCs();
+
+		Assert.assertTrue(uncategorizedFolderItemERCs.contains("linkedERC"));
+		Assert.assertTrue(uncategorizedFolderItemERCs.contains("customERC"));
+	}
+
+	private String _getExportedObjectFolderJSONObject(long objectFolderId)
```

Comment:
I believe we can inline the `_getExportedObjectFolderJSONObject(emptyFolder.getId())` because we are only using it once.